esp-wifi/src/ble/btdm.rs

```diff
@@ -177,9 +177,8 @@ unsafe extern "C" fn queue_delete(queue: *const ()) {
     trace!("queue_delete {:?}", queue);
 
     let queue = queue as *mut ConcurrentQueue;
-    (*queue).release_storage();
     unsafe {
-        crate::compat::malloc::free(queue.cast());
+        core::ptr::drop_in_place(queue);
```

Comment:
I think `drop_in_place` just drops the contents of the pointer, but won't deallocate it, so the `free` is still necessary at this moment.

Also, since these functions are 1:1 the same as `compat::common::create/delete_queue`, can we reuse them here?

Reply:
Indeed - the docs weren't too clear about that to me - or I missed it 👍